homeassistant/components/dwd_weather_warnings/coordinator.py

```diff
@@ -4,23 +4,61 @@
 
 from dwdwfsapi import DwdWeatherWarningsAPI
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+from .const import (
+    CONF_REGION_DEVICE_TRACKER,
+    CONF_REGION_IDENTIFIER,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+)
+from .exceptions import EntityNotFoundError
+from .util import get_position_data
 
 
 class DwdWeatherWarningsCoordinator(DataUpdateCoordinator[None]):
     """Custom coordinator for the dwd_weather_warnings integration."""
 
-    def __init__(self, hass: HomeAssistant, api: DwdWeatherWarningsAPI) -> None:
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize the dwd_weather_warnings coordinator."""
         super().__init__(
             hass, LOGGER, name=DOMAIN, update_interval=DEFAULT_SCAN_INTERVAL
         )
 
-        self.api = api
+        self._api = DwdWeatherWarningsAPI(None)
+        self._device_tracker = None
+
+    async def async_config_entry_first_refresh(self) -> None:
+        """Perform first refresh."""
+        if region_identifier := self.config_entry.data.get(CONF_REGION_IDENTIFIER):
+            self._api = await self.hass.async_add_executor_job(
+                DwdWeatherWarningsAPI, region_identifier
+            )
+        else:
+            self._device_tracker = self.config_entry.data.get(
+                CONF_REGION_DEVICE_TRACKER
+            )
+
+        await super().async_config_entry_first_refresh()
 
     async def _async_update_data(self) -> None:
         """Get the latest data from the DWD Weather Warnings API."""
-        await self.hass.async_add_executor_job(self.api.update)
+        if self._device_tracker:
+            try:
+                position = get_position_data(self.hass, self._device_tracker)
+            except (EntityNotFoundError, AttributeError) as err:
+                raise UpdateFailed(f"Error fetching position: {repr(err)}") from err
+
+            self._api = await self.hass.async_add_executor_job(
+                DwdWeatherWarningsAPI, position
+            )
+        else:
+            if self._api is None:
```

Comment:
Can this happen? `self._api` is unconditionally assigned in `__init__`

Reply:
It might not be possible anymore. I think ruff marked this line at one point but I can't check for around 3 weeks as I'm on vacation as of today.

Will check after that of course

Reply:
OK. I've marked the PR as draft, please mark it ready for review when you've had time to check 👍

Reply:
Removed it in df914c2d60aae7b7ccc3193cdb0e7a74cf3254f1 :)